pkg/update/check.go

```diff
@@ -0,0 +1,119 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+type versionInfo struct {
+	LatestVersion string    `json:"latest_version"`
+	LatestDate    time.Time `json:"latest_date"`
+}
+
+type announcement struct {
+	FromDate     time.Time `json:"from_date"`
+	ToDate       time.Time `json:"to_date"`
+	Announcement string    `json:"announcement"`
+}
+
+type updateResponse struct {
+	Trivy         versionInfo    `json:"trivy"`
+	Announcements []announcement `json:"announcements"`
+	Warnings      []string       `json:"warnings"`
+}
+
+var (
+	updatesApi       = "https://api.trivy.cloud/check"
+	responseRecieved atomic.Bool
+	currentVersion   string
+	latestVersion    updateResponse
+)
+
+// CheckUpdate makes a best efforts request to determine the
+// latest version and any announcements
+func CheckUpdate(ctx context.Context, version string, args []string) {
+	currentVersion = version
+
+	go func() {
+		args = getFlags(args)
```

Comment:
does `https://api.trivy.cloud/check` show info only about enabled deprecated flags?
Or why we need to send info about args.

BTW we already shows warnings about deprecated flags:
https://github.com/aquasecurity/trivy/blob/dd54f80d3fda7821dba13553480e9893ba8b4cb3/pkg/flag/options.go#L122-L124

Reply:
The example about the deprecated flag was more about showing what could be included in the payload - version, warnings, announcements etc - I know we add warnings about deprecations in the top of the output. 

This was more showing that we could notify resistant users that in the nth release it would be removed entirely